fit/curve_nd_test.go

```diff
@@ -0,0 +1,189 @@
+// Copyright ©2020 The go-hep Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package fit_test
+
+import (
+	"bufio"
+	//"image/color"
+	"log"
+	"math/rand"
+	"os"
+	"strconv"
+	"strings"
+	"testing"
+
+	"go-hep.org/x/hep/fit"
+	"go-hep.org/x/hep/hbook"
+	"go-hep.org/x/hep/hplot"
+	"gonum.org/v1/gonum/floats"
+	"gonum.org/v1/gonum/optimize"
+	"gonum.org/v1/plot/plotter"
+	"gonum.org/v1/plot/vg"
+)
+
+func TestCurve2D(t *testing.T) {
+	ExampleCurve2D_plane()
+}
+
+func ExampleCurve2D_plane() {
+	var (
+		m1    = 0.3
+		m2    = 0.1
+		c     = 0.2
+		ps    = []float64{m1, m2, c}
+		n0    = 30
+		n1    = 30
+		x0min = -1.
+		x0max = 1.
+		x1min = -1.
+		x1max = 1.
+	)
+
+	plane := func(x, ps []float64) float64 {
+		return ps[0]*x[0] + ps[1]*x[1] + ps[2]
+	}
+
+	xdata, ydata := genData2D(n0, n1, plane, ps, x0min, x0max, x1min, x1max)
+
+	res, err := fit.CurveND(
+		fit.FuncND{
+			F: func(x []float64, ps []float64) float64 {
+				return plane(x, ps)
+			},
+			X: xdata,
+			Y: ydata,
+			N: 3,
+		},
+		nil, &optimize.NelderMead{},
+	)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	if err := res.Status.Err(); err != nil {
+		log.Fatal(err)
+	}
+	if got, want := res.X, []float64{m1, m2, c}; !floats.EqualApprox(got, want, 0.1) {
+		log.Fatalf("got= %v\nwant=%v\n", got, want)
+	}
+
+	{
+		p := hplot.New()
```

Comment:
it seems the plot image you've attached doesn't completely correspond to this code.
(at least the title doesn't match what's in that code)